atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/transaction/impl/SnapshotTransaction.java

```diff
@@ -1969,16 +1969,12 @@ private void commitWrites(TransactionService transactionService) {
             // This must happen before conflict checking, otherwise we could complete the checks and then have someone
             // else write underneath us before we proceed (thus missing a write/write conflict).
             // Timing still useful to distinguish bad lock percentiles from user-generated lock requests.
-            LockToken commitLocksToken = timedAndTraced("commitAcquireLocks", this::acquireLocksForCommit);
-            closer.register(() -> {
-                // Not timed because tryUnlock() is an asynchronous operation.
-                traced("postCommitUnlock", () -> timelockService.tryUnlock(ImmutableSet.of(commitLocksToken)));
-            });
+            LockToken commitLockToken = transactionLocksManager.registerLock(
+                    timedAndTraced("commitAcquireLocks", this::acquireLocksForCommit));
 
             // Conflict checking. We can actually do this later without compromising correctness, but there is no
             // reason to postpone this check - we waste resources writing unnecessarily if these are going to fail.
-            timedAndTraced(
-                    "commitCheckingForConflicts", () -> throwIfConflictOnCommit(commitLocksToken, transactionService));
+            timedAndTraced("commitCheckingForConflicts", () -> throwIfConflictOnCommit(transactionService));
```

Comment:
In this call, we'll source the locks from the manager correctly.

Reply:
Agreed.